### PR TITLE
Bug/add legacy support

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #23 
- #25 

**This PR will result in the following new package version:**

- v0.7.0 due to schema changes

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Features
> - For Fivetran Lever connectors created on or after July 27, 2024, the `USER` and `INTERVIEWER_USER` source tables have been renamed to `USERS` and `INTERVIEW_USER`, respectively. This package now prioritizes the `USERS` and `INTERVIEW_USER` tables if available, falling back to `USER` and `INTERVIEWER_USER` if not.
>   - To continue using the `USER` and/or `INTERVIEWER_USER` tables, set the variables `lever__using_users` and/or `lever__using_interview_user` to false in your `dbt_project.yml`.
>   - For more information, refer to the [July 2024 connector release notes](https://fivetran.com/docs/connectors/applications/lever/changelog#july2024) and the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md##leveraging-legacy-connector-table-names).
> 
> - Introduced the ability to union source data from multiple Lever connectors. For more details, see the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md#union-multiple-connectors).
> 
> ## Bug fixes
> - Fixed an issue where the dbt package would error due to a missing `CONTACT_LINK` source table for users without source data, even though it was enabled in the Fivetran Connector. A null-filled table will now be generated in such cases.
> 
> ## Under the hood
> - Updated temporary models to union source data using the `fivetran_utils.union_data` macro.
> - Added the `source_relation` column in each staging model to identify the origin of each field, utilizing the `fivetran_utils.source_relation` macro.
> - Updated tests to include the new `source_relation` column.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [ ] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- Similar updates as found in [THIS](https://github.com/fivetran/dbt_recharge_source/pull/12) Recharge PR.

### Missing contact_link
- Simulated `CONTACT_LINK` as a missing source, and the null table is properly created.
Screenshot 2024-10-21 at 5 36 15 PM

### `user` vs `users` and `interviewer_user` vs `interview_user`
- Include both versions as seeds to test both versions of each source. Snippet from run_models.sh
```bash
dbt run --target "$db" --full-refresh
dbt test --target "$db"
dbt run --vars '{lever__using_users: false, lever__using_interview_user: false, lever_using_posting_tag: false, lever_using_requisitions:false}' --target "$db" --full-refresh
dbt test --vars '{lever__using_users: false, lever__using_interview_user: false, lever_using_posting_tag: false, lever_using_requisitions:false}' --target "$db"
```

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
 name: 'lever_source'
-version: '0.6.0'
+version: '0.7.0'
 config-version: 2
 require-dbt-version: [">=1.3.0", "<2.0.0"]
 vars:
-
-  lever_using_posting_tag: true
-  lever_using_requisitions: true
```
Comment (fivetran-catfritz) [inline dbt_project.yml:8]: These are true by default, so should not be included here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+  - To continue using the `USER` and/or `INTERVIEWER_USER` tables, set the variables `lever__using_users` and/or `lever__using_interview_user` to false in your `dbt_project.yml`.
+  - For more information, refer to the [July 2024 connector release notes](https://fivetran.com/docs/connectors/applications/lever/changelog#july2024) and the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md##leveraging-legacy-connector-table-names).
+
+- Introduced the ability to union source data from multiple Lever connectors. For more details, see the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md#union-multiple-connectors).
+
+## Bug fixes
+- Fixed an issue where the dbt package would error due to a missing `CONTACT_LINK` source table for users without source data, even though it was enabled in the Fivetran Connector. A null-filled table will now be generated in such cases.
```
Comment (fivetran-reneeli) [inline CHANGELOG.md:12]:
```suggestion
- Fixed an issue where the dbt package would error due to a missing `CONTACT_LINK` source table for users without the titular source data, even if it was enabled in the Fivetran Connector. A null-filled table will now be generated in such cases.
```

File: integration_tests/dbt_project.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ vars:
     lever_requisition_offer_identifier: "requisition_offer"
     lever_requisition_identifier: "requisition"
     lever_stage_identifier: "stage"
-    lever_user_identifier: "user"
+    lever_users_identifier: "users"
```
Comment (fivetran-reneeli) [inline integration_tests/dbt_project.yml:27]: Shall we also include `lever_interviewer_user_identifier` and `lever_user_identifier` to test as well? we can comment it out, but i think it is worthwhile to keep. Also, I noticed it's retained in the transforms branch.

**Review by fivetran-reneeli — COMMENTED** [review]
Thanks @fivetran-catfritz ! Just had some comments

**Review by fivetran-catfritz — COMMENTED** [review]
Thanks @fivetran-reneeli! I have updated.

**Review by fivetran-reneeli — APPROVED** [review]
Just remembered one thing-- to uncollapse the dropdowns in the README. Other than that, looks good to go!

File: models/tmp/stg_lever__interviewer_user_tmp.sql [hunk 4]
```diff
@@ -1 +1,12 @@
...
+        table_identifier='interviewer_user', 
+        database_variable='lever_database', 
+        schema_variable='interview_user' if var('lever__using_interview_user', lever_source.does_table_exist('interview_user')) else 'interviewer_user', 
+        default_database=target.database,
+        default_schema='lever',
+        default_variable='interview_user' if var('lever__using_interview_user', lever_source.does_table_exist('interview_user')) else 'interviewer_user',
+        union_schema_variable='lever_union_schemas',
+        union_database_variable='lever_union_databases'
+    )
+}}
```
Comment (fivetran-joemarkiewicz) [inline models/tmp/stg_lever__interviewer_user_tmp.sql:12]: For some reason this isn't working as I would expect when testing on two different schemas (one with `interviewer_user` and one with `interview_user`. It seems to work when `interviewer_user` is present, but not `interview_user`. 

Strangely enough, the `users` vs `user` works without an issue. Would you be able to look into this and see what could be the cause or if I may be missing something.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz thanks for working through this PR! I have a few change requests and comments before this is ready to go. Let me know if you have any questions. Thanks!

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz Thanks for the review. Made the updates!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz I added the null tests back!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks @fivetran-catfritz! Approved with one last request to add an additional entry to the CHANGELOG for the test modification/updates.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+  - For more information, refer to the [July 2024 connector release notes](https://fivetran.com/docs/connectors/applications/lever/changelog#july2024) and the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md##leveraging-legacy-connector-table-names).
+- Introduced the ability to union source data from multiple Lever connectors. For more details, see the related [README section](https://github.com/fivetran/dbt_lever_source/blob/main/README.md#union-multiple-connectors).
+
+## Bug fixes
+- Fixed an issue where the dbt package would error due to a missing `CONTACT_LINK` source table for users without the titular source data, even if it was enabled in the Fivetran Connector. A null-filled table will now be generated in such cases.
+
+## Under the hood
+- Turned off freshness tests for `USER`/`USERS` and `INTERVIEW_USER`/`INTERVIEWER_USER` to avoid possible conflicts.
+- Updated temporary models to union source data using the `fivetran_utils.union_data` macro.
+- Added the `source_relation` column in each staging model to identify the origin of each field, utilizing the `fivetran_utils.source_relation` macro.
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply CHANGELOG.md:16]: Added!

**Pull request merged**